addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/internal/LoxoneHandlerFactory.java

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.loxone.internal;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import org.eclipse.smarthome.config.discovery.DiscoveryServiceRegistry;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.type.ChannelGroupType;
+import org.eclipse.smarthome.core.thing.type.ChannelGroupTypeUID;
+import org.eclipse.smarthome.core.thing.type.ChannelType;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeProvider;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.eclipse.smarthome.io.transport.upnp.UpnpIOService;
+import org.openhab.binding.loxone.LoxoneBindingConstants;
+import org.openhab.binding.loxone.handler.LoxoneMiniserverHandler;
+
+import com.google.common.collect.Sets;
+
+/**
+ * Factory responsible for creating Loxone things (Miniservers) and their handlers ({@link LoxoneMiniserverHandler}
+ *
+ * @author Pawel Pieczul - Initial contribution
+ */
+public class LoxoneHandlerFactory extends BaseThingHandlerFactory implements ChannelTypeProvider {
+    private List<ChannelType> channelTypes = new CopyOnWriteArrayList<>();
+    private List<ChannelGroupType> channelGroupTypes = new CopyOnWriteArrayList<>();
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Sets
+            .newHashSet(LoxoneBindingConstants.THING_TYPE_MINISERVER);
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    @Override
+    protected ThingHandler createHandler(Thing thing) {
+        ThingTypeUID uid = thing.getThingTypeUID();
+        if (uid.equals(LoxoneBindingConstants.THING_TYPE_MINISERVER)) {
+            LoxoneMiniserverHandler handler = new LoxoneMiniserverHandler(thing, this);
```

Comment:
Agree. You may also want to look at Harmony Hub binding, which I took as an example, it has the same error (and channel type provider mixed into handler factory).